### PR TITLE
Www 17 Staff Picks Routing

### PR DESCRIPTION
**Pull request by holingpoon**
Will work with @ktp242 on resolving conflicts in `monthData.js`, and figure out branch merging scheme. Thanks!

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Fix the merge conflict and try to fix the tests too. The broken tests are not related to your work, but they should still work given your update.

File: src/app/routes/routes.jsx [hunk 1]
```diff
@@ -4,26 +4,23 @@ import React from 'react';
 import { IndexRoute, Route, Redirect } from 'react-router';
 
 // Import components
-import App from '../components/Application/Application';
-import Main from '../components/Application/Main';
-import BookPage from '../components/BookPage/BookPage';
-import Error404Page from '../components/Error404Page/Error404Page';
+import App from '../components/Application/Application.jsx';
```
Comment (EdwinGuzman) [inline src/app/routes/routes.jsx]: ESlint complains about having the `.jsx` or `.js` extensions. So let's remove them.

File: src/app/components/ListSelector/ListSelector.jsx [hunk 2]
```diff
@@ -59,42 +59,36 @@ class ListSelector extends React.Component {
...
-        if (response.data.statusCode >= 400) {
+        if (!response.status || response.status >= 400) {
           this.updateBookStore();
-          console.log(
-            `API error with status code ${response.data.statusCode}: ${response.data.errorMessage}`
-          );
+          console.log(`API error with status code ${response.status}: ${response.data.errorMessage}`);
           // Leads the user to the 404 page
-          this.updateLocation('/books-music-dvds/recommendations/staff-picks/404');
+          this.updateLocation('/books-music-movies/recommendations/best-books/404');
```
Comment (EdwinGuzman) [inline src/app/components/ListSelector/ListSelector.jsx]: Use the constant variables from `appConfig` here.

File: src/app/components/ListSelector/ListSelector.jsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ class ListSelector extends React.Component {
     currentSeason = '',
     listType = 'staff-picks',
     filters = [],
-    selectedFilters = []
+    selectedFilters = [],
```
Comment (ktp242) [inline src/app/components/ListSelector/ListSelector.jsx]: This should not have the trailing `,` .It is one of the arguments of a function.

File: src/app/components/ListSelector/ListSelector.jsx [hunk 4]
```diff
@@ -59,42 +59,36 @@ class ListSelector extends React.Component {
...
           // Leads the user to the 404 page
-          this.updateLocation('/books-music-dvds/recommendations/staff-picks/404');
+          this.updateLocation(`${config.baseUrl}404`);
         } else {
           // For valid API response, updates BookStore for the new list
           this.updateBookStore(
             response.data.currentPicks,
             submitValue,
-            'staff-picks'
+            'staff-picks',
```
Comment (ktp242) [inline src/app/components/ListSelector/ListSelector.jsx]: This should not have the trailing , .It is one of the arguments of a function.

**Comment by EdwinGuzman** [comment]
@ktp242 eslint complains about the trailing comma because it a list. It doesn't matter if it's an array and there's a list there, or if they're function params, they're still lists.

**Comment by holingpoon** [comment]
@EdwinGuzman So should there be commas after the last item? Should they stay off for best practices?

**Comment by EdwinGuzman** [comment]
If we're following eslint, yes, the commas should be there. Also, the local import statements should not have an extension at the end (`jsx`, `js`);

**Comment by holingpoon** [comment]
Okay, @ktp242 @EdwinGuzman I kept the trailing commas. I also removed `.jsx` extensions from the `import` statements.

**Comment by ktp242** [comment]
I remember once I ran into a broken function because a trailing `,` attached to the arguments. Just want to make sure it doesn't break anything.

**Comment by holingpoon** [comment]
@ktp242 That is also my concern, `localhost` seems to be tolerant of trailing commas right now. Maybe deploying to `development` will give us better clues?

**Comment by EdwinGuzman** [comment]
When you deploy, the app will be serving the translated and compiled JS which will remove the commas I think (when babel does the transpilation). Locally, the app is serving the jsx files so if it works locally then it should work when deployed, but it's go to check either way.

**Comment by EdwinGuzman** [comment]
Works well.

File: src/server/routes/monthData.js [hunk 5]
```diff
@@ -52,10 +53,10 @@ function currentMonthData(req, res, next) {
 
       next();
     })
-    .catch(error => {
+    .catch((error) => {
       console.error(`Status Code: ${error.statusCode}, Error Message: ${error.code}`);
 
-      return res.redirect('/books-music-dvds/recommendations/404');
+      return res.redirect('/books-music-movies/recommendations/404');
```
Comment (gkallenberg) [inline src/server/routes/monthData.js]: We're not using config.baseUrl here

File: src/server/routes/monthData.js [hunk 6]
```diff
@@ -90,60 +91,59 @@ function selectMonthData(req, res, next) {
   if (!seasonMatches || !isValidAudience) {
     console.error('Status Code: 400, Error Message: Invalid season or audience.');
 
-    return; res.redirect('/books-music-dvds/recommendations/404');
+    return; res.redirect('/books-music-movies/recommendations/404');
```
Comment (gkallenberg) [inline src/server/routes/monthData.js]: We're not using config.baseUrl here

File: src/server/routes/monthData.js [hunk 7]
```diff
@@ -90,60 +91,59 @@ function selectMonthData(req, res, next) {
...
+
+        // Calls the selected list
+        return nyplApiClientGet(`${platformConfig.endpoints.staffPicksPath}${requestedSeason}`);
+      })
+      .then(data => {
+        // If error returned from the endpoint
+        if (data.statusCode >= 400) {
+          console.error(`Status Code: ${data.statusCode}, Error Message: ${data.error}`);
+
+          return res.redirect('/books-music-movies/recommendations/404');
```
Comment (gkallenberg) [inline src/server/routes/monthData.js]: We're not using config.baseUrl here

File: src/server/routes/monthData.js [hunk 8]
```diff
@@ -194,15 +193,17 @@ function selectMonthDataFormPost(req, res) {
   const audienceQuery = audience ? `?audience=${audience}` : '';
 
   if (!season || !audience) {
+
     console.error(
       `Form data of season or audience is undefined. season: ${season}, audience: ${audience}`
     );
 
-    res.redirect('/books-music-dvds/recommendations/404');
+    res.redirect('/books-music-movies/recommendations/404');
```
Comment (gkallenberg) [inline src/server/routes/monthData.js]: We're not using config.baseUrl here

File: src/server/routes/monthData.js [hunk 9]
```diff
@@ -90,60 +91,59 @@ function selectMonthData(req, res, next) {
...
+          },
+          pageTitle: '',
+          metaTags: [],
+        };
+        next();
+      })
+      .catch((error) => {
+        console.error(`Status Code: ${error.statusCode}, Error Message: ${error.code}`);
+
+        return res.redirect('/books-music-movies/recommendations/404');
```
Comment (gkallenberg) [inline src/server/routes/monthData.js]: We're not using config.baseUrl here

File: src/server/routes/monthData.js (hunk 5 above)
Comment (EdwinGuzman) [reply src/server/routes/monthData.js]: True, all those redirects should point to the new 404 page, too.

**Pull request merged**